src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1691,5 +1657,41 @@ public int GetHashCode(LibraryRange obj)
                 return combiner.CombinedHash;
             }
         }
+
+        /// <summary>
+        /// Represents a cache key for storing <see cref="ResolvedLibraryEntry" /> items in a dictionary with a unique <see cref="FrameworkRuntimePair" /> and <see cref="LibraryRangeIndex" />.
+        /// </summary>
+        private readonly struct ResolvedLibraryEntryCacheKey : IEquatable<ResolvedLibraryEntryCacheKey>
```

Comment:
Is there an added cost from the constant hashing of FrameworkRuntimePair?
I see that the type is immutable, so they may be an opportunity to improve the performance there by caching the hash.

Reply:
Thanks to this little comment I ended up completely refactoring again from scratch.  Let's chat Monday